src/workloads/docs/ParallelInsert.yml

```diff
@@ -185,6 +185,7 @@ AutoRun:
     mongodb_setup:
     - replica
     - replica-noflowcontrol
+    - replica-all-feature-flags
```

Comment:
This one confused me wiht the PrepareEnvironmentWith. Do I need to do something more here?

Reply:
Hm maybe. I think this is saying "run with mongodb_setup=replica-delay-mixed whenever the existing value of mongodb_stup is replica-all-feature-flags" so I'm not sure if this workload would ever run with mongodb_setup as anything _other_ than replica-delay-mixed. We now include the generated tasks in DSI artifacts so this would be pretty easy to patch-build (just select genny-auto-tasks on the variant and ensure the task is generated on the variant task-gen task)

I proposed a different syntax for this in [TIG-2841](https://jira.mongodb.org/browse/TIG-2841) that may be worth revisiting.

Reply:
I took some time to look through the code and this code is not doing what it is expected to do. I've removed my addition, and the noflowcontrol setup. 

What this was doing: For both the 3-node-replica and 3-node-replica-no-flowcontrol build variants, this would generate two tasks, using replica-delay-mixed and replica mongodb_setups respectively. That is, the no flow control and the default version were running *identical* tests. I think the PrepareEnvironmentWith.mongodbs_setup wants to be a list of lists, with a list per entry in Requires.mongodb_setup. 

I'm queuing up a patch build. Will link it in presently.

Reply:
Thanks for digging. We really didn't know what we wanted it to look like when we added the AutoRun section. I don't think the current syntax is worth augmenting. I'd much rather see something like the proposal in [TIG-2841](https://jira.mongodb.org/browse/TIG-2841) (it's not that much code) which would make what you're after here a lot more intuititive.

Reply:
That sounds fine to me. Thanks Ryan. The patch build ran by the way. I didn't check it closely, but I did see that it ran the genny tasks.  https://evergreen.mongodb.com/version/6050cec132f4177273d509f1